.github/workflows/ci_linting.yml

```diff
@@ -0,0 +1,35 @@
+---
+name: Linters
+
+on:
+  pull_request:
+    branches:
+      - master
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        command:
+          - |
+            echo "cppcheck"
+            source ./codebuild/bin/s2n_setup_env.sh
```

Comment:
This is repeated.  Seems simple, why not put this in its own step?

Reply:
iirc, each step is a separate bash process ?

Reply:
An overhaul of these bash scripts is valid feedback, but probably out of scope for now...

Reply:
Environment variables set in one step are not automatically carried over to subsequent steps. You have to explicitly tell GHA which ones to keep around[1]. Until we can overhaul the setup scripts, this will need to be done this way. :/

[1] https://help.github.com/en/actions/reference/development-tools-for-github-actions#set-an-environment-variable-set-env